lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1315,7 +1295,7 @@ def deploy_vm(self):
         self.configure_cpu_and_memory(vm_obj=vm_obj, vm_creation=True)
         self.configure_disks(vm_obj=vm_obj)
         self.configure_network(vm_obj=vm_obj)
-        if self.should_deploy_from_template() and len(self.params['customize']) > 0:
+        if vm_obj is None or self.should_deploy_from_template():
```

Comment:
in fact, you are wrong @aperigault because if you add that you will break the guest_map customizations which are handled in customize_vm and doesn't require customize variable namespace